elfi/visualization/visualization.py

```diff
@@ -271,9 +271,13 @@ def progress_bar(iteration, total, prefix='', suffix='', decimals=1, length=100,
         Bar fill character
 
     """
-    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
-    filled_length = int(length * iteration // total)
-    bar = fill * filled_length + '-' * (length - filled_length)
-    print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
-    if iteration == total:
-        print()
+    if total > 0:
+        percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+        filled_length = int(length * iteration // total)
+        bar = fill * filled_length + '-' * (length - filled_length)
+        print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
+        if iteration == total:
+            print()
+    else:
+        print("Can't output progress bar. Number of total iterations is zero. \n"
```

Comment:
I don't think this else block is needed. The progress bar is not called by the user, so getting a message like this is just confusing.